tests/unit/type.test.ts

```diff
@@ -0,0 +1,20 @@
+import type from '../../src/decorators/type'
+import { Model } from '../../src'
+
+describe('@type() decorator', () => {
+  test('basic decorators', () => {
+    class Cat extends Model {
+      @type('string') str: string
+      @type('number') num: string
+      @type('date') dat: string
+      @type('boolean') bol: string
+    }
+
+    expect(Cat.meta.attributeDefinition).toEqual({
+      str: 'string',
+      num: 'number',
+      dat: 'date',
+      bol: 'boolean'
+    })
+  })
+})
```

Comment:
The `type` decorator simply turns the type decoration into model metadata so that the model is ready to be `setup` which involves looking at the metadata for a model and setting up attrs, relations, adapters, serializers etc.